src/reducers/vms.js

```diff
@@ -15,9 +15,19 @@ function removeVms ({ state, payload: { vmIds } }) {
   return mutable.asImmutable()
 }
 
-function updateVmDisk ({ state, payload: { vmId, disk } }) {
+function removeMissingVms ({ state, payload: { vmIdsToPreserve } }) {
+  const vmIdsToBeDeleted = state.get('vms')
+    .keySeq()
+    .filter(vmId => !vmIdsToPreserve.some(toPreserveId => toPreserveId === vmId))
```

Comment:
please consider using a `Set` instead of doing nested linear lookups to check for presence